File: InterviewBit/Array/AddOneToNumber/addOneToNumber.py
class Solution:
    # @param A : list of integers
    # @return a list of integers
    def plusOne(self, A):
        A.insert(0,0)
        A[-1] += 1
        if A[-1]==10:
            n = len(A) - 1
            while (A[n]==10):
                A[n] = 0
                n=n-1
                A[n] += 1
        while (A[0]==0):
            A.pop(0)
        return A

File: InterviewBit/Array/AddOneToNumber/addOneToNumber.cpp
vector<int> Solution::plusOne(vector<int> &A) {
    A.insert(A.begin(),0);
    A[A.size()-1]++;
    for(int i=A.size()-1; i>=0; i--){
        if(A[i]>9) A[i]%=10, A[i-1]++;
    }
    while(A[0]==0) A.erase(A.begin());
    return A;
}